### DISCUSSION TITLE
API reference feature like Javadoc

### DISCUSSION THREAD

**Discussion by jimlyas** [squidfunk/mkdocs-material#2697] [Ideas]
I'm starting building a library for android, and it's sure a lot of fun using mkdocs-material for documentation like initial configuration and features of the library. But i started to wonder to implement API reference for each class of the library using mkdocs-material but i don't find it anywhere.

Is it possible to use mkdocs-material as API reference like this:
![image](https://user-images.githubusercontent.com/33854739/119429553-43033000-bd39-11eb-87ac-ae5a05a81eee.png)

Currently, because it's Android Kotlin library... I'm using [dokka](https://github.com/Kotlin/dokka) which return HTML file.

**Comment by squidfunk** [comment]
If a JavaDoc generator could generate Markdown, then you can certainly use Material for MkDocs to render this Markdown. Other than that, MkDocs is a plain and simple static site generator. It doesn't understand any programming language by itself.

**Comment by sandydunlop** [comment]
I ended up in the same situation a few months ago and all the javadoc to Markdown tools I could find were last updated years ago, so I ended up creating a doclet to output markdown with MkDocs in mind.

It's own javadoc is an example of how it looks with MkDocs:

https://sandydunlop.github.io/markista/javadoc/markista/doclet/MarkdownDoclet/

**Reply by pawamoy** [reply]
I couldn't find the sources for your GitHub Pages, could you show how you inject/render docs for Java code within a MkDocs site? :pray:

**Reply by sandydunlop** [reply]
I'm using an [organization pages](https://www.mkdocs.org/user-guide/deploying-your-docs/#organization-and-user-pages) style setup, so there are two GitHub repos. My MkDocs one was private. I've made it public now so you can [see the Markdown files here](https://github.com/sandydunlop/website/tree/main/docs/markista/javadoc/markista). The files there were produced with the `-tabs` and `--flatten-packages` options.

I run the doclet automatically as part of the Java project build, then have a Gradle task copy the Javadocs into a subdirectory of my MkDocs site.

For the organization style setup, I then deploy the generated site to the GitHub repo for the live webpages.

For project style setups, that last step would be unnecessary.

**Reply by pawamoy** [reply]
Thank you @sandydunlop! :rocket:
